microceph/database/schema.go

```diff
@@ -39,51 +64,109 @@ CREATE TABLE services (
   id                            INTEGER  PRIMARY KEY AUTOINCREMENT NOT NULL,
   member_id                     INTEGER  NOT  NULL,
   service                       TEXT     NOT  NULL,
-  FOREIGN KEY (member_id) REFERENCES "internal_cluster_members" (id) ON DELETE CASCADE,
+  FOREIGN KEY (member_id) REFERENCES "%s" (id) ON DELETE CASCADE,
   UNIQUE(member_id, service)
 );
-  `
+  `, tableName, tableName)
 
-	_, err := tx.Exec(stmt)
+	_, err = tx.Exec(stmt)
 
 	return err
 }
 
 // Adds client config table in database schema.
 func schemaUpdate2(ctx context.Context, tx *sql.Tx) error {
-	stmt := `
+	tableName, err := getClusterTableName(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	stmt := fmt.Sprintf(`
 CREATE TABLE client_config (
   id                            INTEGER  PRIMARY KEY AUTOINCREMENT NOT NULL,
   member_id                     INTEGER,
   key                           TEXT     NOT  NULL,
   value                         TEXT     NOT  NULL,
-  FOREIGN KEY (member_id) REFERENCES "internal_cluster_members" (id) ON DELETE CASCADE
+  FOREIGN KEY (member_id) REFERENCES "%s" (id) ON DELETE CASCADE
 );
 
 CREATE UNIQUE INDEX cc_index ON client_config(coalesce(member_id, 0), key);
-  `
+  `, tableName)
 
-	_, err := tx.Exec(stmt)
+	_, err = tx.Exec(stmt)
 
 	return err
 }
 
 // schemaUpdate3 generates the diskpaths table, copying the data from the disks table.
 func schemaUpdate3(ctx context.Context, tx *sql.Tx) error {
-	stmt := `
+	tableName, err := getClusterTableName(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	stmt := fmt.Sprintf(`
 CREATE TABLE disks2 (
   id                            INTEGER  PRIMARY KEY AUTOINCREMENT NOT NULL,
   member_id                     INTEGER  NOT  NULL,
   path                          TEXT     NOT  NULL,
-  FOREIGN KEY (member_id) REFERENCES "internal_cluster_members" (id) ON DELETE CASCADE,
+  FOREIGN KEY (member_id) REFERENCES "%s" (id) ON DELETE CASCADE,
   UNIQUE(member_id, path)
 );
 INSERT INTO disks2 (id, member_id, path)
 SELECT osd, member_id, path FROM disks;
 DROP TABLE disks;
 ALTER TABLE disks2 RENAME TO disks;
+  `, tableName)
+	_, err = tx.Exec(stmt)
+
+	return err
+}
+
+// schemaUpdate4 updates all tables referencing `internal_cluster_members` to now reference `core_cluster_members` instead.
+func schemaUpdate4(ctx context.Context, tx *sql.Tx) error {
+	stmt := `
+CREATE TABLE disks2 (
```

Comment:
@sabaini It would be good to have a discussion for table names being introduced here.

Reply:
FWIW, the tables in MicroCeph have not been renamed, it is only the foreign key reference to the microcluster table that has changed. We ensure that foreign key references are not erased when we changed the table name internally, so no data will be lost. 

This had to be retroactively applied to all schema updates in case a very old version of MicroCeph tries to update to the latest version.